test/irb/test_tracer.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: false
+require 'tempfile'
+require 'irb'
+require 'rubygems' if defined?(Gem)
+
+require_relative "helper"
+
+module TestIRB
+  class ContextWithTracerTest < TestCase
+    def setup
+      IRB.init_config(nil)
+      IRB.conf[:VERBOSE] = false
+    end
+
+    def test_use_tracer_is_disabled_by_default
+      IRB.conf[:USE_TRACER] = false
+      workspace = IRB::WorkSpace.new(Object.new)
+      irb = IRB::Irb.new(workspace, TestInputMethod.new)
+      # Context still needs this setter to be defined
+      assert irb.context.respond_to?(:use_tracer=)
+    end
+
+    def test_use_tracer_when_gem_is_unavailable
+      IRB.conf[:USE_TRACER] = true
+      begin
+        gem 'tracer'
+        omit "Skipping test_use_tracer_when_gem_is_unavailable because 'tracer' gem is available."
+      rescue Gem::LoadError
```

Comment:
IMO, in general a test case shouldn't rescue any error to test the error handling logic as it only captures part of the behaviour while trying to mimic the rest. Unfortunately we currently don't have a good infra for this kind of dependency-dependant tests. I think we can leave this case untested for now.

Reply:
This is required to skip this test when `USE_TRACER=true` is set, but run it when it is not set / false.